### DISCUSSION TITLE
My repository only identifies Jinja and ignores other languages

### DISCUSSION THREAD

**Discussion by rw-rw-rw-fermino** [github-linguist/linguist#6202] [Classification] [answered]
I made an ansible + vagrant project, with the following languages:
- YAML(in tasks and playbooks)
- INI (in ansible.cfg file)
- Jinja2 (in two lines of code)
- Ruby(in Vagrantfile)

Linguist is detecting that 100% of the project is Jinja, I tried using a .gitattributes file but it didn't help.

My repo: https://github.com/fermino-linux/3-node-swarm

**Comment by lildude** [comment, marked as answer]
> * YAML(in tasks and playbooks)
> * INI (in ansible.cfg file)

YAML and INI is considered `data` so ignored by default. If you want them to counted and for YAML to show in the side bar, you need to use the `linguist-detectable` override.

> * Jinja2 (in two lines of code)

Already detected, so nothing to do here.

> * Ruby(in Vagrantfile)

It is already being correctly detected as Ruby without an override, but Linguist considered it as [vendored](https://github.com/github/linguist/blob/c1c34e5260797b4d598f5ec76f19723bfc5a1894/lib/linguist/vendor.yml#L354) so you'll need to use the `-linguist-vendored` override if you want this to appear in your sidebar.

More info on the overrides and links to the files affecting things can be found in https://github.com/github/linguist/blob/master/docs/overrides.md
